simulation/traffic_simulator/test/src/behavior/test_longitudinal_speed_planner.cpp

```diff
@@ -0,0 +1,354 @@
+
+  const double target_speed = 8.5;
+  const double expected_duration = 4.0;
+
+  const double result_duration =
+    planner.getAccelerationDuration(target_speed, constraints, current_twist, current_accel);
+  EXPECT_NEAR(result_duration, expected_duration, tolerance);
+}
+
+/**
+ * @note Test functionality aggregation used in other classes; test function behavior
+ * when target_speed is bigger than max speed.
+ */
+TEST(LongitudinalSpeedPlanner, getDynamicStates_targetSpeedOverLimit)
+{
+  const auto planner = makeLongitudinalSpeedPlanner();
+  const auto current_twist = makeTwistWithLinearX(10.0);
+  const auto current_accel = makeAccelWithLinearX(1.0);
+  const auto constraints =
+    traffic_simulator_msgs::build<traffic_simulator_msgs::msg::DynamicConstraints>()
+      .max_acceleration(1.0)
+      .max_acceleration_rate(1.0)
+      .max_deceleration(1.0)
+      .max_deceleration_rate(1.0)
+      .max_speed(20.0);
+
+  double target_speed = 100.0;
+  const auto [result0_twist, result0_accel, result0_jerk] =
+    planner.getDynamicStates(target_speed, constraints, current_twist, current_accel);
+
+  target_speed = constraints.max_speed;
+  const auto [result1_twist, result1_accel, result1_jerk] =
+    planner.getDynamicStates(target_speed, constraints, current_twist, current_accel);
+
+  EXPECT_ACCEL_EQ(result0_accel, result1_accel);
+  EXPECT_TWIST_EQ(result0_twist, result1_twist);
+  EXPECT_DOUBLE_EQ(result0_jerk, result1_jerk);
+}
+
+/**
+ * @note Test functionality aggregation used in other classes; test calculations correctness
+ * with target_speed = max speed and current speed = 0 - goal is to test the situation
+ * when acceleration is so long that max jerk is reached.
+ */
+TEST(LongitudinalSpeedPlanner, getDynamicStates_maxJerk)
+{
+  const auto planner = makeLongitudinalSpeedPlanner();
+  const auto current_twist = makeTwistWithLinearX(0.0);
+  const auto current_accel = makeAccelWithLinearX(0.0);
+  auto constraints =
+    traffic_simulator_msgs::build<traffic_simulator_msgs::msg::DynamicConstraints>()
+      .max_acceleration(5.0)
+      .max_acceleration_rate(1.0)
+      .max_deceleration(5.0)
+      .max_deceleration_rate(1.0)
+      .max_speed(20.0);
+
+  const double target_speed = constraints.max_speed;
+
+  const double result0_jerk =
+    std::get<2>(planner.getDynamicStates(target_speed, constraints, current_twist, current_accel));
+  EXPECT_DOUBLE_EQ(result0_jerk, constraints.max_acceleration_rate);
+
+  constraints.max_acceleration_rate = 2.0;
+  const double result1_jerk =
+    std::get<2>(planner.getDynamicStates(target_speed, constraints, current_twist, current_accel));
+  EXPECT_DOUBLE_EQ(result1_jerk, constraints.max_acceleration_rate);
+}
+
+/**
+ * @note Test functionality aggregation used in other classes; test calculations correctness
+ * with target_speed slightly smaller than current speed - goal is to test the situation
+ * when the max jerk is not reached because the target_speed is reached first.
```

Comment:
Please change line. Do not split sentence by `;`

```suggestion
 * @note Test functionality aggregation used in other classes
 * Test calculations correctness with target_speed slightly smaller than current speed - goal is to test the situation when the max jerk is not reached because the target_speed is reached first.
```

Reply:
Please correct the other sections in the same manner.